docs/user-guide/onEnter.md

```diff
@@ -0,0 +1,48 @@
+# onEnter
+
+What do we want to do when we enter the `Loading` state? 
+We want to make the http request to load the items from our server, right?
+Let's specify that with the DSL in our state machine:
+
+```kotlin
+class ItemListStateMachine(
+    private val httpClient: HttpClient
+) : FlowReduxStateMachine<ListState, Action>(initialState = Loading) {
+
+    init {
+        spec {
+            inState<Loading> {
+                onEnter { state: State<Loading> ->
+                    // we entered the Loading state,
+                    // so let's do the http request
+                    try {
+                        val items = httpClient.loadItems()  // loadItems() is a suspend function
+                        state.override { ShowContent(items) }  // return ShowContent from onEnter block
+                    } catch (t: Throwable) {
+                        state.override { Error(t) }   // return Error state from onEnter block
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+There are a some new things like  `onEnter` and `State<T>`.
+We will covered `State<T>` in the next section.
+
+Let's talk about `onEnter`:
+
+- **`onEnter { ... }` is running asynchronously in a coroutine**. That means whatever you do inside the `onEnter` block
+  is not blocking anything else. You can totally run here long-running and expensive calls (like doing a http request).
```

Comment:
Just a small thing but technically if you make a blocking call here (blocking as in blocking a thread, not just a suspending function call) this will actually block the main thread.

Reply:
How would you change it? To " long running suspending" calls?

Reply:
Yes, replacing blocking with suspending would resolve it. We could have something about actual blocking calls which would be fine if you do them in these blocks with a `withContext(Dispatchers.IO)` but I think that goes too far and is probably more confusing then helpful. (plus it's a general coroutines thing)